lib/stat_tracker.rb

```diff
@@ -182,71 +182,79 @@ def lowest_scoring_home_team
   end
 
   # SEASON STATISTICS
+  # season stats helper methods -------------------
   def season_games(season)
     games.find_all { |game| game.season == season }
   end
+  #-----------------------------
 
   def winningest_coach(season)
-    #season_games_by_id returns an array of just season game ids
     season_game_ids = season_games(season).map do |game|
       game.game_id
     end
 
-    #then find all games in game_teams from the above season
     season_game_teams = game_teams.find_all do |game|
       season_game_ids.include?(game.game_id)
     end
 
-    # filter season games by wins
-    season_wins = season_game_teams.find_all { |game| game.result == "WIN" }
+    game_teams_by_coach = season_game_teams.group_by do |game_team|
+      game_team.head_coach
+    end
+
+    coach_games_and_wins = Hash.new { |h,k| h[k] = Hash.new(0) }
```

Comment:
I wonder if using reduce here could help make 204-209 a tiny bit more concise?

Reply:
since I haven't been as deep in in it with your set of stats methods, it'd take me a bit of time to come up with a concrete suggestion for how to implement it and i could be missing an obvious reason as to why it doesn't work. but just something to come back to in refactoring since this method is quite long